src/matcha_ml/state/matcha_state.py

```diff
@@ -2,9 +2,56 @@
 import hashlib
 import json
 import os
+from dataclasses import dataclass
 from typing import Dict, List, Optional
 
 
+@dataclass
+class MatchaResource:
+    """A class to represent a resource in the state."""
+
+    name: str
+
+
+@dataclass
+class MatchaResourceProperty:
+    """A class to represent a resource property in the state."""
+
+    name: str
+    value: str
+
+
+@dataclass
+class MatchaStateComponent:
+    """A class to represent a component in the state."""
+
+    resource: MatchaResource
+    properties: List[MatchaResourceProperty]
+
+
+@dataclass
+class MatchaState:
```

Comment:
Can/should we also add a `__dict__` method for users interacting with `MatchaState`?

Reply:
`__dict__` is used for introspection rather than a function for the user to interact with the object itself. I think we want to be more explicit here, i.e., `to_dict()` - this is a clearer way for the user to get a dictionary representation of the object